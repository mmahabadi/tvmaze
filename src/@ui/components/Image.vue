<script setup lang="ts">
import { defineProps } from 'vue'

defineProps<{ src?: string; alt?: string }>()
</script>
<template>
  <img v-if="src" :src="src" :alt="alt" class="w-full h-full object-cover" />
  <img
    v-else
    class="w-full h-full object-cover"
    src="@/assets/image-placeholder.jpg"
    alt="image-placeholder"
  />
</template>
