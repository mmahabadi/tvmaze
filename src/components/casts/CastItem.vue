<script setup lang="ts">
import { type Person } from '@/types'
import Image from '@/@ui/components/Image.vue'

type propType = Pick<Person, 'name' | 'image'>

defineProps<propType>()
</script>
<template>
  <div class="cast-item">
    <Image :src="image" :alt="name" class="img" />
    <p class="text-center">{{ name }}</p>
  </div>
</template>
<style scoped>
.cast-item {
  @apply w-full p-3 rounded flex-col bg-gray-900 border border-gray-800;

  .img {
    @apply w-full h-96 object-cover rounded mb-4;
  }
}
</style>
