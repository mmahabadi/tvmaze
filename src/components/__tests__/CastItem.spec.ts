import { describe, it, expect, beforeEach } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'
import CastItem from '../casts/CastItem.vue'
import Image from '@/@ui/components/Image.vue'

describe('Component', () => {
  let wrapper: VueWrapper<CastItem>
  const props = {
    name: 'Mohsen',
    image: 'mohsen.jpg'
  }

  beforeEach(() => {
    wrapper = mount(CastItem, {
      props,
      global: {
        components: {
          Image
        }
      }
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders details', () => {
    expect(wrapper.find('p.text-center').text()).toBe(props.name)
    expect(wrapper.findComponent(Image).props('src')).toBe(props.image)
    expect(wrapper.findComponent(Image).props('alt')).toBe(props.name)
  })
})
