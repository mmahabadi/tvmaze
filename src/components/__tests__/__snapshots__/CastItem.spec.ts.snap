// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component > renders correctly 1`] = `
"<div data-v-6563c4e7="" class="cast-item"><img data-v-6563c4e7="" src="mohsen.jpg" alt="Mohsen" class="w-full h-full object-cover img">
  <p data-v-6563c4e7="" class="text-center">Mohsen</p>
</div>"
`;
